athena/modules/text/module_text_llm/module_text_llm/cot_prev_submission/generate_suggestions.py

```
@@ -0,0 +1,144 @@
+    get_chat_prompt_with_formatting_instructions,
+    check_prompt_length_and_omit_features_if_necessary,
+    num_tokens_from_prompt,
+)
+from llm_core.utils.predict_and_parse import predict_and_parse
+from module_text_llm.helpers.utils import add_sentence_numbers, get_index_range_from_line_range, \
+    format_grading_instructions
+from module_text_llm.cot_prev_submission.prompt_analyzer import SubmissionComparisonResult
+from module_text_llm.cot_prev_submission.prompt_generate_feedback import AssessmentModel
+
+
+async def generate_suggestions(exercise: Exercise, submission: Submission, config: ApproachConfig, *, debug: bool,
```

Pylint (3.3.7)

[refactor] Too many arguments (7/5)

(R0913)